ui/package.json

```diff
@@ -26,8 +26,9 @@
     "web3modal": "^1.9.8"
   },
   "scripts": {
+    "contract-dependencies": "(cd ../contract && yarn && yarn run build)",
     "start": "react-scripts start",
-    "build": "react-scripts build",
+    "build": "yarn contract-dependencies && react-scripts build",
```

Comment:
as I know `build` is only used for prod, to do local development we use `start`

when I was talking about the script I mean like just a shell script which will do compile of contracts, typechain generation and put types to ui folder

Reply:
just read your description :D
I would say I have doubts that Spheron will manage to compile contracts from an external folder as well

Reply:
Yes, just experimented with a few different option for Spheron and seems like it cannot make it.
I see three options here:
* Drop Spheron usage
* Find Spheron analogue which could make it
* Keep commiting generated typechain

For hackathon I would continue with last one to not drop Spheron. But if we decide to continue with the project after that - ofc need to refactor

Reply:
yep, I think the last option for hackathon is fine

Reply:
Updated PR: kept scripts for manual trigger + used `yarn`